test/k8sT/DatapathConfiguration.go

```diff
@@ -645,9 +645,86 @@ var _ = Describe("K8sDatapathConfig", func() {
 			}, DeployCiliumOptionsAndDNS)
 			Expect(testPodConnectivityAcrossNodes(kubectl)).Should(BeTrue(), "Connectivity test between nodes failed")
 		})
+
+		SkipItIf(helpers.RunsOnGKE, "With VXLAN", func() {
+			deploymentManager.DeployCilium(map[string]string{
+				"hostFirewall": "true",
+			}, DeployCiliumOptionsAndDNS)
+			testHostFirewall(kubectl)
+		})
+
+		// We need to skip this test when using kube-proxy because of #14859.
+		SkipItIf(helpers.RunsWithKubeProxy, "With native routing", func() {
+			options := map[string]string{
+				"hostFirewall":         "true",
+				"tunnel":               "disabled",
+				"autoDirectNodeRoutes": "true",
+			}
+			// We can't rely on gke.enabled because it enables
+			// per-endpoint routes which are incompatible with
+			// the host firewall.
+			if helpers.RunsOnGKE() {
+				options["gke.enabled"] = "false"
+			}
+			deploymentManager.DeployCilium(options, DeployCiliumOptionsAndDNS)
+			testHostFirewall(kubectl)
+		})
 	})
 })
 
+func testHostFirewall(kubectl *helpers.Kubectl) {
+	randomNs := deploymentManager.DeployRandomNamespaceShared(DemoHostFirewall)
+	deploymentManager.WaitUntilReady()
+
+	demoHostPolicies := helpers.ManifestGet(kubectl.BasePath(), "host-policies.yaml")
+	By(fmt.Sprintf("Applying policies %s", demoHostPolicies))
+	_, err := kubectl.CiliumPolicyAction(randomNs, demoHostPolicies, helpers.KubectlApply, helpers.HelperTimeout)
+	ExpectWithOffset(1, err).Should(BeNil(), fmt.Sprintf("Error creating resource %s: %s", demoHostPolicies, err))
+
+	defer func() {
+		By(fmt.Sprintf("Removing policies %s", demoHostPolicies))
+		_, err = kubectl.CiliumPolicyAction(randomNs, demoHostPolicies, helpers.KubectlDelete, helpers.HelperTimeout)
+		ExpectWithOffset(1, err).Should(BeNil(), fmt.Sprintf("Error removing resource %s: %s", demoHostPolicies, err))
+	}()
+
+	By("Checking host policies on ingress from local pod")
+	testHostFirewallWithPath(kubectl, randomNs, "zgroup=testClient", "zgroup=testServerHost", false)
```

Comment:
I was under the impression that host firewall policies only apply to pods deployed in the host namespace? I don't follow the motivation behind introducing these pod<-> pod (deployed in non-host ns) tests.

Reply:
The pod selected by `zgroup=testServerHost` is deployed in host network namespace.

Reply:
So what's the significance of `randomNS` here?

Reply:
For example the Cilium pods are in host-network namespace and can be found in `kube-system` or `cilium` Kubernetes namespaces.

Reply:
Got it. Thanks!